mlflow/utils/model_utils.py

```diff
@@ -122,7 +126,18 @@ def _validate_and_copy_code_paths(code_paths, path, default_subpath="code"):
     if code_paths is not None:
         code_dir_subpath = default_subpath
         for code_path in code_paths:
-            _copy_file_or_tree(src=code_path, dst=path, dst_dir=code_dir_subpath)
+            try:
+                _copy_file_or_tree(src=code_path, dst=path, dst_dir=code_dir_subpath)
+            except OSError as e:
+                raise MlflowException(
+                    message=(
+                        f"Failed to copy the specified code path '{code_path}' into the model "
+                        "artifacts. It appears that your code path includes file(s) that cannot "
+                        "be copied, such as Databricks Notebooks. Please specify a code path "
```

Comment:
This is a databricks-specific error,

what about checking `is_in_databricks()` firstly, if True then we raise this error ? Otherwise it might confuse OSS users.

Reply:
Nice idea! Will add a flag and switch error message accordingly.